pkg/cmd/roachtest/roachtestutil/validation_check.go

```diff
@@ -46,13 +50,15 @@ func CheckReplicaDivergenceOnDB(ctx context.Context, l *logger.Logger, db *gosql
 	// TODO(erikgrinaker): avoid result set buffering. We seem to be receiving
 	// results in batches of 64 rows, regardless of results_buffer_size or the
 	// row size (e.g. with 16 KB ballast per row). Not clear where this buffering
-	// is happening or how to disable it.
+	// is happening or how to disable it. This also means that we can take more
+	// than maxInconsistencies storage checkpoints, if they happen within the
+	// same buffered result.
 	started := timeutil.Now()
 	rows, err := db.QueryContext(ctx, `
 SET CLUSTER SETTING kv.consistency_queue.testing_fast_efos_acquisition.enabled = true;
 SET statement_timeout = '20m';
 SELECT t.range_id, t.start_key_pretty, t.status, t.detail
-FROM crdb_internal.check_consistency(false, '', '') as t;`)
+FROM crdb_internal.check_consistency(false, '', '', true) as t;`)
```

Comment:
On second thought, this is a bit tricky since we'll likely backport the change, so we'll have to check for a patch version which we don't really know yet.

The consistency check will just log an error if this should happen, without failing the test -- maybe that's fine?

https://github.com/cockroachdb/cockroach/blob/f17c6a849f3a6de8d0204adb4f46025e0f3a6538/pkg/cmd/roachtest/roachtestutil/validation_check.go#L70-L76

Reply:
Huh interesting.

> Ideally this would be rock solid.

Do you know if things have changed since this comment was added? I wonder what value we get from just logging the error; in all likelihood, legitimate errors would go unnoticed in CI runs.